pkg/repo/repo.go

```diff
@@ -127,36 +159,270 @@ func CreateRepoWithMetadata(ctx context.Context, targets []TargetWithMetadata) (
 //   - `rekor` = it will get Usage set to `Rekor`
 //   - `tsa` = it will get Usage set to `tsa`.
 //   - Anything else will get set to `Unknown`
-func CreateRepo(ctx context.Context, files map[string][]byte) (tuf.LocalStore, string, error) {
-	targets := make([]TargetWithMetadata, 0, len(files))
+//
+// The targets will be added individually to the TUF repo if CreateRepoOptions.AddMetadataTargets
+// is set to true. The trusted_root.json file will be added if CreateRepoOptions.AddTrustedRoot
+// is set to true. At least one of these has to be true.
+func CreateRepoWithOptions(ctx context.Context, files map[string][]byte, options CreateRepoOptions) (tuf.LocalStore, string, error) {
+	if !options.AddMetadataTargets && !options.AddTrustedRoot {
+		return nil, "", errors.New("failed to create TUF repo: At least one of metadataTargets, trustedRoot must be true")
+	}
+
+	metadataTargets := make([]TargetWithMetadata, 0, len(files))
 	for name, bytes := range files {
-		var usage string
-		switch {
-		case strings.Contains(name, "fulcio"):
-			usage = "Fulcio"
-		case strings.Contains(name, "ctfe"):
-			usage = "CTFE"
-		case strings.Contains(name, "rekor"):
-			usage = "Rekor"
-		case strings.Contains(name, "tsa"):
-			usage = "TSA"
-		default:
-			usage = "Unknown"
-		}
-		scmActive, err := json.Marshal(&sigstoreCustomMetadata{Sigstore: CustomMetadata{Usage: usage, Status: "Active"}})
+		scmActive, err := json.Marshal(&sigstoreCustomMetadata{Sigstore: CustomMetadata{Usage: getTargetUsage(name), Status: "Active"}})
 		if err != nil {
 			return nil, "", fmt.Errorf("failed to marshal custom metadata for %s: %w", name, err)
 		}
-		targets = append(targets, TargetWithMetadata{
+		metadataTargets = append(metadataTargets, TargetWithMetadata{
 			Name:           name,
 			Bytes:          bytes,
 			CustomMetadata: scmActive,
 		})
 	}
 
+	targets := make([]TargetWithMetadata, 0, len(files)+1)
+	if options.AddMetadataTargets {
+		targets = append(targets, metadataTargets...)
+	}
+	if options.AddTrustedRoot {
+		trustedRootTarget, err := constructTrustedRoot(metadataTargets)
+		if err != nil {
+			return nil, "", fmt.Errorf("failed to construct trust root: %w", err)
+		}
+		targets = append(targets, *trustedRootTarget)
+	}
+
 	return CreateRepoWithMetadata(ctx, targets)
 }
 
+func constructTrustedRoot(targets []TargetWithMetadata) (*TargetWithMetadata, error) {
```

Comment:
Can we move this to sigstore-go? There is some overlap with https://github.com/sigstore/cosign/issues/3794, and I'd prefer to have this centrally located right off the bat.

@cmurphy FYI, since you were asking about utilities for trust root generation as well.